vscode-sbcl-clpm.md

```diff
@@ -0,0 +1,472 @@
+---
+title: Getting started with SBCL and CLPM using VSCode
+---
+
+This page is meant to provide an introduction to using
+[VSCode](https://code.visualstudio.com/) as a Lisp IDE with
+[SBCL](http://sbcl.org) as the Common Lisp implementation and
+[CLPM](https://clpm.dev) as the package manager.  Please feel free to
+reach me at hello at smashedtoatoms dot com if you run into trouble.
+
+![](assets/commonlisp-vscode-alive.png)
+
+### Prerequisites And Explanations
+
+- [VsCode](https://code.visualstudio.com) with [command
+  line](https://code.visualstudio.com/docs/setup/mac#_launching-from-the-command-line)
+  installed
+- [Alive](https://marketplace.visualstudio.com/items?itemName=rheller.alive)
+  Common Lisp plugin for VsCode
+- [SBCL](http://www.sbcl.org)
+- [CLPM](https://www.clpm.dev)
+- [rlwrap](https://github.com/hanslub42/rlwrap)
+
+#### Why Use SBCL?
+
+You can honestly use whatever you want. The nice thing about Common Lisp
+being a standard is all of the implementations will all run your code
+pretty much the same. They all have slightly different goals, and each
+is a little different where the standard is not defined, but you can
+deal with it when you encounter it. I use SBCL because it is updated
+regularly, it is fast, and it is easy. If you're new to Common Lisp,
+just start with it. You can change it out later when you have your arms
+around the whole Common Lisp thing.
+
+#### Why use CLPM?
+
+CLPM has [very well-defined goals](https://www.clpm.dev/#project-goals).
+It doesn't force the consumers of your code to use it. It's written in
+Common Lisp, so it's easy to hack on if you feel so inclined.
+
+CLPM gives you deterministic builds via version pinning in projects with
+no other dependencies. It allows different projects on your computer to
+pin different versions of the same library. It has a lockfile! It's
+common in the Common Lisp world to share dependencies that update once a
+month across projects because everything is stable. I prefer more
+granular control.  [Qlot](https://github.com/fukamachi/qlot) can do this
+for you as well, though it is tricky to use without
+[Roswell](https://github.com/roswell/roswell).
```

Comment:
but there is no explanation in relation to the VSCode setup? Is it *necessary* to use CLPM?

These are nice but probably don't belong here.

Reply:
It isn't necessary to use clpm.  This whole section was me trying to give someone coming from something like rust or js a workflow that was more familiar, while also giving them a repl and demonstrate why the CL debugger is so neat.

The biggest hurdle I had when I started writing Common Lisp was making sense of the tooling.  I was trying to provide some instructions that would get someone set up with a base project with tests and a working repl in which they could experiment and would start to show them what's possible while making it so they don't have to fully grok the entire environment.

All that being said, I will remove referencing the other things.

Reply:
Do you think your explanations could be put earlier / somewhat in a (new?) Cookbook page and thus be useful to newcomers? 

I was thinking a "CL features" page could demonstrate some stuff. We could speak about the ecosystem and give comparisons.